game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/aa/FiringGroupSplitterAa.java

```diff
@@ -0,0 +1,80 @@
+package games.strategy.triplea.delegate.battle.steps.fire.aa;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ACTIVE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ALIVE;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.TechAbilityAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.fire.FiringGroup;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Creates AA and Targeted Hit firing groups
+ *
+ * <p>The firing groups are separated by typeAa and isSuicideOnHit
+ */
+@Value(staticConstructor = "of")
+public class FiringGroupSplitterAa implements Function<BattleState, List<FiringGroup>> {
+
+  BattleState.Side side;
+
+  @Override
+  public List<FiringGroup> apply(final BattleState battleState) {
```

Comment:
The original logic that this is based off of is in `MustFightBattle#updateOffensiveAaUnits` (https://github.com/triplea-game/triplea/blob/e160d341f6c60ecf5297aad9ab26ac5a524dc2dd/game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java#L928), `MustFightBattle#updateDefensiveAaUnits` (https://github.com/triplea-game/triplea/blob/e160d341f6c60ecf5297aad9ab26ac5a524dc2dd/game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java#L949), `FireAa#FireAa` (https://github.com/triplea-game/triplea/blob/e160d341f6c60ecf5297aad9ab26ac5a524dc2dd/game-core/src/main/java/games/strategy/triplea/delegate/battle/FireAa.java#L66), and `FireAa#execute` (https://github.com/triplea-game/triplea/blob/e160d341f6c60ecf5297aad9ab26ac5a524dc2dd/game-core/src/main/java/games/strategy/triplea/delegate/battle/FireAa.java#L100).